layer1/executor/tasks/tests/snapshot_test.go

```diff
@@ -0,0 +1,450 @@
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 0
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 1
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 2
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 3
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 4
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 5
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 6
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	//blocksSinceDesperation = 1
+	eth.GetCurrentHeightFunc.SetDefaultReturn(uint64(12), nil)
+	groupSignatureHash = common.HexToHash("0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470")
+
+	iValidator = 1
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.False(t, result)
+
+	iValidator = 2
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 3
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.True(t, result)
+
+	iValidator = 4
+	result, err = utils.AmILeading(eth, ctx, logger, start, groupSignatureHash.Bytes(), nValidator, iValidator, desperationFactor, desperationDelay)
+	assert.Nil(t, err)
+	assert.False(t, result)
+}
+
+func getRandomNumberInRange(min, max int) int {
+	rand.Seed(time.Now().UnixNano())
+	return rand.Intn(max-min) + min
+}
+
+func GenerateSnapshotData(chainID uint32, height uint32, bnSigners []*crypto.BNGroupSigner, n int, mpkI []*big.Int, fakeSig bool) ([]byte, []byte, error) {
```

Comment:
Anywhere we're doing test helper functions it can often times make code cleaner if we pass in a `*testing.T` and just `t.Fatal`/`t.Error` on any errors instead of returning them. Less error handling required on the calling tests.

Reply:
sure thing!